add linux documentation (#3401)
DataDog/documentation: content/tracing/setup/dotnet.md

.NET Core doesn't support the Windows Global Assembly Cache (GAC) used by the .NET Framework. Instead, you can deploy Datadog's managed assemblies together with your application by adding the `Datadog.Trace.ClrProfiler.Managed` NuGet package. It is important that you match the versions of the NuGet package and the MSI installer.

For example, if you are using the MSI installer for version `0.5.0-beta`, use:

```
dotnet add package Datadog.Trace.ClrProfiler.Managed --version 0.5.0-beta
```

### Linux

Automatic instrumentation is available on Linux for .NET Core 2.0+. To set up automatic instrumentation, 4 steps are necessary:

#### Install the Datadog Agent

> Traces are sent to a local Datadog Agent and then forwarded to Datadog. To install the agent, follow the [Linux instructions][1].

Reviewer: ```suggestion
Traces are sent to a local Datadog Agent and then forwarded to Datadog. To install the Agent, follow the [Linux instructions][1].
```

Reviewer 2: Good catch. This PR was already merged, so I slipped this change into #3409.

```bash
dotnet add package Datadog.Trace.ClrProfiler.Managed --version 0.5.0-beta
```

#### Install the `Datadog.Trace.ClrProfiler.Native` Library

Automatic instrumention in Linux is implemented using a shared library available from the `dd-trace-csharp` [releases page][10].

For Debian or Ubuntu, download and install the Debian package:

```bash
curl -LO https://github.com/DataDog/dd-trace-csharp/releases/download/v0.5.0-beta/datadog-dotnet-apm_0.5.0_amd64.deb
sudo apt install ./datadog-dotnet-apm_0.5.0_amd64.deb
```

For CentOS or Fedora, download and install the RPM package

```bash
curl -LO https://github.com/DataDog/dd-trace-csharp/releases/download/v0.5.0-beta/datadog-dotnet-apm-0.5.0-1.x86_64.rpm
sudo rpm -Uvh datadog-dotnet-apm-0.5.0-1.x86_64.rpm
```

A Tar archive is available for other distributions:

```bash
sudo mkdir -p /opt/datadog
curl -L https://github.com/DataDog/dd-trace-csharp/releases/download/v0.5.0-beta/datadog-dotnet-apm-0.5.0.tar.gz \
| sudo tar xzf - -C /opt/datadog
```

For Alpine Linux you will also need to install `libc6-compat`

```bash
apk add libc6-compat
```

> 

Reviewer: This might be a good place to use tabs. [See here for reference](https://github.com/DataDog/documentation/blob/f885a792aed60860ec6522b0486e6160e55bdc56/content/agent/faq/agent-commands.md#L20)

.NET Core doesn't support the Windows Global Assembly Cache (GAC) used by the .NET Framework. Instead, you can deploy Datadog's managed assemblies together with your application by adding the `Datadog.Trace.ClrProfiler.Managed` NuGet package. It is important that you match the versions of the NuGet package and the MSI installer.

For example, if you are using the MSI installer for version `0.5.0-beta`, use:

```
dotnet add package Datadog.Trace.ClrProfiler.Managed --version 0.5.0-beta
```

### Linux

> Automatic instrumention is available on Linux for .NET Core 2.0+. To setup automatic instrumentation, 4 steps are necessary:

Reviewer 2: - typo in "instrumention"
- "setup" (one word) is a noun or adjective, "to set up" is a verb

```suggestion
Automatic instrumentation is available on Linux for .NET Core 2.0+. To set up automatic instrumentation, 4 steps are necessary:
```

> 1. [Install the Datadog Agent][1]

Reviewer 2: The code blocks are breaking up the numbered list items so they show up as 1, 2, 1, and 1 (see [preview](https://docs-staging.datadoghq.com/caleb/dotnet-linux/tracing/setup/dotnet/)). I don't know if there's a way around this except to change them to bullet points.

Author: it's proper markdown. I guess our setup doesn't understand it. I will change it to use sub-headers instead.

> 1. Add the `Datadog.Trace.ClrProfiler.Managed` nuget package to your project:

Reviewer 2: ```suggestion
1. Add the `Datadog.Trace.ClrProfiler.Managed` NuGet package to your project:
```

   ```bash
   curl -LO https://github.com/DataDog/dd-trace-csharp/releases/download/v0.5.0-beta/datadog-dotnet-apm-0.5.0-1.x86_64.rpm
   sudo rpm -Uvh datadog-dotnet-apm-0.5.0-1.x86_64.rpm
   ```

   A Tar archive is available for other distributions:

   ```bash
   sudo mkdir -p /opt/datadog
   curl -L https://github.com/DataDog/dd-trace-csharp/releases/download/v0.5.0-beta/datadog-dotnet-apm-0.5.0.tar.gz \
   | sudo tar xzf - -C /opt/datadog
   ```

   For Alpine Linux you will also need to install `libc6-compat`

   ```bash
   apk add libc6-compat
   ```

1. Add environment variables to your service to enable the .NET tracer.

   For Systemd:

   ```ini
    [Unit]
    Description=example

    [Service]
    ExecStart=/usr/bin/dotnet /app/example.dll
    Restart=always
    # Datadog .NET Tracer Environment Variables
    Environment=CORECLR_ENABLE_PROFILING=1
    Environment=CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
    Environment=CORECLR_PROFILER_PATH=/opt/datadog/Datadog.Trace.ClrProfiler.Native.so
    Environment=DD_INTEGRATIONS=/opt/datadog/integrations.json

    [Install]
    WantedBy=multi-user.target
   ```

   For Docker:

   ```docker

>    ENV CORECLR_ENABLE_PROFILING=1

Reviewer 2: This is drawing weird red lines before and after some lines (see [preview](https://docs-staging.datadoghq.com/caleb/dotnet-linux/tracing/setup/dotnet/)). Should we drop the `docker` formatting?
